pkg/component/component.go

```diff
@@ -402,6 +339,215 @@ func getS2IPaths(podEnvs []corev1.EnvVar) []string {
+	cmpMaxCPU := componentConfig.GetMaxCPU()
+	cmpMinMemory := componentConfig.GetMinMemory()
+	cmpMaxMemory := componentConfig.GetMaxMemory()
+	cmpSrcRef := componentConfig.GetRef()
+	appName := componentConfig.GetApplication()
+
+	log.Successf("Initializing '%s' component", cmpName)
+	createArgs := occlient.CreateArgs{
+		Name:            cmpName,
+		ImageName:       cmpType,
+		ApplicationName: appName,
+	}
+	createArgs.SourceType = cmpSrcType
+	createArgs.SourcePath = componentConfig.GetSourceLocation()
+
+	if len(cmpPorts) > 0 {
+		createArgs.Ports = cmpPorts
+	}
+
+	if cmpMaxCPU != "" && cmpMinCPU != "" {
+		createArgs.Resources = append(
+			createArgs.Resources,
+			*(util.FetchResourceQuantity(corev1.ResourceCPU, cmpMinCPU, cmpMaxCPU, "")),
+		)
+	}
+	if cmpMaxMemory != "" && cmpMinMemory != "" {
+		createArgs.Resources = append(
+			createArgs.Resources,
+			*(util.FetchResourceQuantity(corev1.ResourceMemory, cmpMinMemory, cmpMaxMemory, "")),
+		)
+	}
+	switch cmpSrcType {
+	case config.GIT:
+		// Use Git
+		if cmpSrcRef != "" {
+			createArgs.SourceRef = cmpSrcRef
+		}
+		if err = CreateFromGit(
+			client,
+			createArgs,
+		); err != nil {
+			return errors.Wrapf(err, "failed to create component with args %+v", createArgs)
+		}
+		// Git is the only one using BuildConfig since we need to retrieve the git
+		if err = Build(client, createArgs.Name, createArgs.ApplicationName, true, stdout); err != nil {
+			return errors.Wrapf(err, "failed to build component with args %+v", componentConfig)
+		}
+	case config.LOCAL:
+		fileInfo, err := os.Stat(createArgs.SourcePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get info of path %+v of component %+v", createArgs.SourcePath, createArgs)
+		}
+		if !fileInfo.IsDir() {
+			return fmt.Errorf("component creation with args %+v as path needs to be a directory", createArgs)
+		}
+		// Create
+		if err = CreateFromPath(client, createArgs); err != nil {
+			return errors.Wrapf(err, "failed to create component with args %+v", createArgs)
+		}
+	case config.BINARY:
+		if err = CreateFromPath(client, createArgs); err != nil {
+			return errors.Wrapf(err, "failed to create component with args %+v", createArgs)
+		}
+	default:
+		// If the user does not provide anything (local, git or binary), use the current absolute path and deploy it
+		createArgs.SourceType = config.LOCAL
+		dir, err := os.Getwd()
+		if err != nil {
+			return errors.Wrapf(err, "cannot create component with current path as local source path since no component source details are passed")
+		}
+		createArgs.SourcePath = dir
+		if err = CreateFromPath(client, createArgs); err != nil {
+			return errors.Wrapf(err, "")
+		}
+	}
+	return
+}
+
+// ValidateComponentCreateRequest validates request for component creation and returns errors if any
+// Parameters:
+//	componentSettings: Component settings
+//	isCmpExistsCheck: boolean to indicate whether or not error out if component with same name already exists
```

Comment:
the comment regarding the returned items?

Reply:
Ack